templates/repo/header.tmpl

```diff
@@ -1,38 +1,53 @@
 <div class="header-wrapper">
 {{with .Repository}}
 	<div class="ui container">
-		<div class="repo-header">
+		<div class="repo-header fw">
 			<div class="repo-title-wrap df fc">
-				<div class="repo-title">
+				<div class="repo-title fw">
 					{{$avatar := (repoAvatar . 32 "mr-3")}}
-					{{if $avatar}}
+					{{if $.PageIsSettings}}
+						<div class="repo-icon mr-3">{{svg "octicon-tools" 32}}</div>
+					{{else if $avatar}}
 						{{$avatar}}
 					{{else}}
 						{{template "repo/icon" .}}
 					{{end}}
-					<a href="{{AppSubUrl}}/{{.Owner.Name}}">{{.Owner.Name}}</a>
-					<div class="mx-2">/</div>
-					<a href="{{$.RepoLink}}">{{.Name}}</a>
-					<div class="labels df ac fw">
-						{{if .IsTemplate}}
-							{{if .IsPrivate}}
-								<span class="ui basic label">{{$.i18n.Tr "repo.desc.private_template"}}</span>
-							{{else}}
-								{{if .Owner.Visibility.IsPrivate}}
-									<span class="ui basic label">{{$.i18n.Tr "repo.desc.internal_template"}}</span>
+					<div class="df fw">
+						<a href="{{AppSubUrl}}/{{.Owner.Name}}">{{.Owner.Name}}</a>
+						<div class="mx-2">/</div>
+					</div>
+					<div class="di">
+						<a href="{{$.RepoLink}}">{{.Name}}</a>
+						{{- if $.PageIsSettings -}}
```

Comment:
It may be helpful to add `/* */` within the `{{- ... -}}` to explain that the `{{-` is intentional and why it is intentional.

Reply:
Looked around in the code base and found several instances of  `/* */`. I'm not sure whether I need to explain the intention here. The immediate parent 2 lines above clearly says `di` (`display: inline !important;`).

Reply:
A lot of our UI stuff is essentially being done by cargo-cult - it's not clear why stuff is needed or why it breaks when you adjust the templates. We need to get better and if there is something unusual it needs to be explained. 

If there is a necessary thing like this which could break the UI - it needs to be explicit - otherwise the next person who comes along will add another block in there without the `{{-` whitespace removal and then wonder why it doesn't work.

Reply:
Yes I understand that. A lot of the spacing is controlled by whitespace (I even saw a `&nbsp;&nbsp;&nbsp;`), and to me the margin and padding convenience classes work like whitespace (or one might say `\hspace` from LaTeX) since regular whitespaces are eaten by flexboxes.

Reply:
How do you like this?

Also, found this from you ;-)
![whitespace](https://user-images.githubusercontent.com/1714243/109071479-0ef58000-76f4-11eb-9bc4-44912082f81c.png)